cdc/sink/codec/interface.go

```diff
@@ -44,3 +49,37 @@ type EventBatchDecoder interface {
 	// NextDDLEvent returns the next DDL event if exists
 	NextDDLEvent() (*model.DDLEvent, error)
 }
+
+// Protocol is the protocol of the mq message
+type Protocol int
+
+// Enum types of the Protocol
+const (
+	ProtocolDefault Protocol = iota
+	ProtocolCanal
+)
+
+// FromString converts the protocol from string to Protocol enum type
+func (p *Protocol) FromString(protocol string) {
+	switch strings.ToLower(protocol) {
+	case "default":
```

Comment:
I think `json` is more appropriate, and rename `ProtocolDefault` to `ProtocolJson`
```suggestion
	case "json":
```

Reply:
Maybe `json` is not accurate here, in fact the json encoder is a batch implement of TiCDC open protocol.